pkg/engine/exceptions.go

```diff
@@ -63,6 +65,17 @@ func matchesException(
 		)
 		// if there's no error it means a match
 		if err == nil {
+			if rule.HasValidatePodSecurity() && candidate.HasPodSecuruty() {
+				// copy the rule
+				r := rule.DeepCopy()
+
+				exclude := make([]kyvernov1.PodSecurityStandard, 1)
+				exclude[0] = *candidate.Spec.PodSecurity
+
+				// Set the validation.podsecurity.exclude in the rule
+				r.Validation.PodSecurity.Exclude = exclude
+				_, _, _ = pss.EvaluatePod(r.Validation.PodSecurity, &v1.Pod{})
+			}
```

Comment:
Pushed up a quick commit to change the logic, can you please have a look at this? @MariamFahmy98